File: 26. Remove Duplicates from Sorted Array.py
# 26. Remove Duplicates from Sorted Array


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i = 0
        seen = set()
        for j in range(len(nums)):
            if nums[j] not in seen:
                nums[i] = nums[j]
                i += 1
                seen.add(nums[j])

        return i

File: 26. Remove Duplicates from Sorted Array.cpp
#include <vector>

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() <= 1) {
            return nums.size();
        }
        
        int i = 1;
        int j = 1;
        while (i < nums.size()) {
            if (nums[i] != nums[j-1]) {
                nums[j] = nums[i];
                i++;
                j++;
            } 
            else {
                i++;
            }
        }
        return j;
    }
};